webapp/docs/add-new-token-to-stake.md

```diff
@@ -0,0 +1,127 @@
+
+```ts
+// webapp/tokenList/stakeTokens.ts
+
+export const stakeWhiteList: Partial<
+  Record<Token['chainId'], Record<string, StakeExtensions>>
+> = {
+    [hemi.id]: {
+        ...
+        // satUSD
+        '0xb4818BB69478730EF4e33Cc068dD94278e2766cB': {
+            protocol: 'satoshi',
+            priceSymbol: 'usdc',
+            rewards: ['hemi3x', 'satoshi', 'pell'],
+            website: 'https://www.satoshiprotocol.org',
+        },
+        ...
+    },
+    ...
+}
+```
+
+## Step 3 - Add the protocol logo image (if needed)
+
+If the new token protocol is not already in the [protocolImages.ts](../app/[locale]/stake/protocols/protocolImages.ts) file you need to add the protocol logo image to the [/stake/protocols/images](webapp/app/[locale]/stake/protocols/images) folder and add it to the `protocolImages` list as below:
+
+```ts
+// webapp/app/[locale]/stake/protocols/protocolImages.ts
+
+import satoshi from './images/satoshi.png'
+
+...
+
+export const protocolImages: Record<StakeProtocols, StaticImageData> = {
+    ...
+    satoshi,
+    ...
+}
+```
+
+## Step 4 - Add the new rewards icon images (if needed)
+
+If the new token you added have rewards that is not already in the [tokenRewards.tsx](../app/[locale]/stake/_components/tokenRewards.tsx) list you must add the icon image file to the [rewards icons](webapp/app/[locale]/stake/_components/icons) folder and create the reward points tag component as following:
+
+```ts
+// webapp/app/[locale]/stake/_components/pointsTag.tsx
+
+import { SatoshiPointsIcon } from './icons/satoshiPointsIcon'
+
+...
+
+export const SatoshiPoints = () => (
+  <PointsTag
+    backgroundColor="bg-black"
+    icon={<SatoshiPointsIcon />}
+    label="points"
+    textColor="text-white"
+  />
+)
+```
+
+After that you just need to add the reward points tag component to the [tokenRewards.tsx](../app/[locale]/stake/_components/tokenRewards.tsx) list:
+
+```ts
+// webapp/app/[locale]/stake/_components/tokenRewards.tsx
+
+import {
+  ...
+  SatoshiPoints,
+  ...
+} from './pointsTag'
+
+const rewardComponentMap: Record<Reward, ReactNode> = {
+  ...
+  satoshi: <SatoshiPoints />,
+  ...
+}
+```
+
+## Step 5 - Check if token is whitelisted
+
+Before merging your changes to the `main` branch you need to make sure the new token you added is already whitelisted to staking and you can do that by calling `tokenAllowlist` method in the [staking pool smart contract](https://explorer.hemi.xyz/address/0x4F5E928763CBFaF5fFD8907ebbB0DAbd5f78bA83) as follows:
+
+```sh
+$ curl -X POST -H "Content-Type: application/json" \
+      -d '{
+          "jsonrpc": "2.0",
+          "id": 1,
+          "method": "eth_call",
+          "params": [
+              {
+                  "to": "0x4F5E928763CBFaF5fFD8907ebbB0DAbd5f78bA83",
+                  "data": "0x8135369a000000000000000000000000b4818bb69478730ef4e33cc068dd94278e2766cb"
```

Comment:
perhaps it's not entirely clear how this `data` parameter is build

Reply:
changed to a Hemi explorer screenshot